cmd/system-probe/main_windows.go

```diff
@@ -8,30 +8,66 @@
 package main
 
 import (
+	"context"
+	"errors"
 	"fmt"
 	"os"
 
-	"golang.org/x/sys/windows/svc"
-
+	"github.com/DataDog/datadog-agent/cmd/agent/common/signals"
 	"github.com/DataDog/datadog-agent/cmd/internal/runcmd"
 	"github.com/DataDog/datadog-agent/cmd/system-probe/command"
+	"github.com/DataDog/datadog-agent/cmd/system-probe/config"
 	"github.com/DataDog/datadog-agent/cmd/system-probe/subcommands"
-	"github.com/DataDog/datadog-agent/cmd/system-probe/windows/service"
+	runsubcmd "github.com/DataDog/datadog-agent/cmd/system-probe/subcommands/run"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/winutil/servicemain"
 )
 
+type service struct {
+	errChan <-chan error
+	ctxChan chan context.Context
+}
+
+func (s *service) Name() string {
+	return config.ServiceName
+}
+
+func (s *service) Init() error {
+	s.ctxChan = make(chan context.Context)
```

Comment:
Where does the context get sent to this channel? `StartSystemProbeWithDefaults` waits for a context to be sent.

Reply:
oops, looks like I forgot to re-test after making the `fxutil.OneShot` change. will fix, thank you.